File: LeetCode_ProblemSet/Problem2843.cpp
#include <string>

class Solution {
   public:
    int countSymmetricIntegers(int low, int high) {
        int cnt = 0;

        for (int num = low; num <= high; num++) {
            std::string s = std::to_string(num);
            if (s.length() % 2 == 1) continue;

            int sum = 0;
            for (int idx = 0; idx < s.length() / 2; idx++) {
                sum += s[idx];
            }
            for (int idx = s.length() / 2; idx < s.length(); idx++) {
                sum -= s[idx];
            }
            cnt += sum == 0;
        }

        return cnt;
    }
};

File: LeetCode_ProblemSet/Rust/src/Problem2843.rs
impl Solution {
    pub fn count_symmetric_integers(low: i32, high: i32) -> i32 {
        let mut cnt = 0;
        for num in low..(high + 1) {
            let s = num.to_string();
            if s.len() % 2 == 1 {
                continue;
            }
            let mut sum = 0;
            for idx in 0..(s.len() / 2) {
                sum += s.chars().nth(idx).unwrap() as i32;
            }
            for idx in (s.len() / 2)..s.len() {
                sum -= s.chars().nth(idx).unwrap() as i32;
            }
            if sum == 0 {
                cnt += 1;
            }
        }
        return cnt;
    }
}